C/plugins/storage/sqlite/common/readings.cpp

```diff
@@ -2255,7 +2255,7 @@ vector<string>  assetCodes;
 		if (rc != SQLITE_OK)
 		{
 			raiseError("purge - phase 3", zErrMsg);
-			sqlite3_free(zErrMsg);
+			// sqlite3_free(zErrMsg); // already freed inside purgeAllReadings, in case of error
```

Comment:
So does this mean we are printing a message from a buffer that has been freed?

Reply:
Yes, seems like I missed it in a hurry to share the code changes with you before leaving office :)

Since the raiseError statement is required with the correct error code somewhere, either I can move the raiseError statement inside purgeAllReadings() or remove the sqlite3_free from inside purgeAllReadings() and restore this code. What makes more sense?

Reply:
Probably the later I would say.

Reply:
I just did that yesterday, please review this PR again.